packages/contracts/contracts/chugsplash/L2ChugSplashDeployer.sol

```diff
@@ -0,0 +1,249 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.5.0 <0.8.0;
+pragma experimental ABIEncoderV2;
+
+/* Library Imports */
+import { Lib_ExecutionManagerWrapper } from "../optimistic-ethereum/libraries/wrappers/Lib_ExecutionManagerWrapper.sol";
+import { Lib_MerkleTree } from "../optimistic-ethereum/libraries/utils/Lib_MerkleTree.sol";
+
+/**
+ * @title L2ChugSplashDeployer
+ */
+contract L2ChugSplashDeployer {
+    
+    /*********
+     * Enums *
+     *********/
+
+    enum ActionType {
+        SET_CODE,
+        SET_STORAGE
+    }
+
+
+    /***********
+     * Structs *
+     ***********/
+
+    struct ChugSplashAction {
+        ActionType actionType;
+        address target;
+        bytes data;
+    }
+
+    struct ChugSplashActionProof {
+        uint256 actionIndex;
+        bytes32[] siblings;
+    }
+
+
+    /*************
+     * Variables *
+     *************/
+
+    // Address that can approve new transaction bundles.
+    address public owner;
+    uint256 public currentBundleNonce;
+    bytes32 public currentBundleHash;
+    uint256 public currentBundleSize;
+    uint256 public currentBundleTxsExecuted;
+    mapping (uint256 => mapping (uint256 => bool)) internal completedBundleActions;
+
+
+    /***************
+     * Constructor *
+     ***************/
+    
+    /**
+     * @param _owner Address that will initially own the L2ChugSplashDeployer.
+     */
+    constructor(
+        address _owner
+    ) {
+        owner = _owner;
```

Comment:
TODO: Add this contract to the state dump process + add a new input variable for "chugsplash owner"

Reply:
Should we we add this TODO inline?  Or is it a blocker for merging this PR?

Reply:
Blocker for merging PR.